solidity/contracts/PhasedEscrow.sol

```diff
@@ -70,26 +70,127 @@ contract PhasedEscrow is Ownable {
+    ) external onlyOwner {
+        address beneficiaryAddress = address(_beneficiary);
+        require(
+            beneficiaryAddress != address(0), 
+            "Beneficiary can not be zero address"
+        );
+        approvedBeneficiaries[beneficiaryAddress] = true;
+        emit BeneficiaryApproved(beneficiaryAddress);
+    }
+
+    /// @notice Returns `true` if the given address has been approved as a
+    ///         beneficiary of the escrow, `false` otherwise.
+    function isBeneficiaryApproved(
+        IBeneficiaryContract _beneficiary
+    ) external view returns (bool) {
+        return approvedBeneficiaries[address(_beneficiary)];
+    }
+
+    /// @notice Transfers the role of drawee to another address. Can be called
+    ///         only by the contract owner.
+    function transferDraweeRole(address newDrawee) public onlyOwner {
+        require(newDrawee != address(0), "New drawee can not be zero address");
+        emit DraweeRoleTransferred(drawee, newDrawee);
+        drawee = newDrawee;
+    }
+
+    /// @notice Withdraws tokens from escrow to selected beneficiaries,
+    ///         transferring to each beneficiary the amount of tokens specified
+    ///         as a parameter. Only beneficiaries previously approved by escrow
+    ///         owner can receive funds.
+    function batchedWithdraw(
+        IBeneficiaryContract[] memory beneficiaries,
+        uint256[] memory amounts
+    ) public onlyDrawee {
+        require(
+            beneficiaries.length == amounts.length,
+            "Mismatched arrays length"
+        );
+
+        for (uint256 i = 0; i < beneficiaries.length; i++) {
+            IBeneficiaryContract beneficiary = beneficiaries[i];
+            require(
+                approvedBeneficiaries[address(beneficiary)],
+                "Beneficiary not approved"
+            );
+            withdraw(beneficiary, amounts[i]);
+        }
+    }
+    function withdraw(IBeneficiaryContract beneficiary, uint256 amount) private  {
+        token.safeTransfer(address(beneficiary), amount);
+        emit TokensWithdrawn(address(beneficiary), amount);
+        beneficiary.__escrowSentTokens(amount);
+    }
 }
 
-/// @title CurveRewardsEscrowBeneficiary
+// Interface representing staking pool rewards contract such as CurveRewards
+// contract deployed for Keep (0xAF379f0228ad0d46bB7B4f38f9dc9bCC1ad0360c) or
+// LPRewards contract from keep-ecdsa repository deployed for Uniswap.
+interface IStakingPoolRewards {
+   function notifyRewardAmount(uint256 amount) external; 
+}
+
+
+/// @title StakingPoolRewardsEscrowBeneficiary
 /// @notice A beneficiary contract that can receive a withdrawal phase from a
 ///         PhasedEscrow contract. Immediately stakes the received tokens on a
-///         designated CurveRewards contract.
-contract CurveRewardsEscrowBeneficiary is Ownable {
+///         designated IStakingPoolRewards contract.
+contract StakingPoolRewardsEscrowBeneficiary is Ownable {
```

Comment:
Can it implement `IBeneficiaryContract` interface?

Reply:
Yes, good point. I will add it here.